### ISSUE TITLE
[Bug]: savefig(..., transparent=True) does not make inset_axes transparent

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#22674]
### Bug summary

All's in the title.

### Code for reproduction

```python
from pylab import *
gca().inset_axes([.1, .2, .3, .4]); savefig("/tmp/test.png", transparent=True)

# then display /tmp/test.png with a viewer that distinguishes between white and transparent: the inset axes is not transparent.
```

### Actual outcome

Inset is not transparent.

### Expected outcome

Inset is transparent (... well, admittedly I guess whether one wants this or not may be figure-dependent?)  (mildly relatedly, see also #10023.)

### Operating system

arch linux

### Matplotlib Version

3.6.0.dev1856+ga2ddfc0c0f

### Matplotlib Backend

agg

### Python version

310

### Jupyter version

ENOSUCHLIB

### Installation

git checkout

**Comment by QuLogic** [comment]
I'm surprised that the regular `Axes` was made transparent in the first place, but apparently that's what it's supposed to do.

**Comment by sramesh750** [comment]
I could start working on this issue!

**Comment by timhoffm** [comment]
@sramesh750 Go for it!

`transparent=True` should make all Axes transparent. Should more special needs arise, we can still add more options like `transparent='figure'` later, but I would hold that until someone asks for it.

**Comment by rcomer** [comment]
Closed by #24816 I think.
